### PR TITLE
[Merged by Bors] - Fix bug where client TLS could not be disabled

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- Client TLS can now be disabled by setting `spec.config.tls: null`
```
spec:
  version: 3.8.0-stackable0.7.1
  config:
    tls: null
    clientAuthentication:
      authenticationClass: zk-client-tls
    quorumTlsSecretClass: tls
```

fixes https://github.com/stackabletech/zookeeper-operator/issues/528

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -566,3 +556,124 @@ pub struct ZookeeperZnodeSpec {
...
+        spec:
+          version: abc
+          config:
+            tls:
+              secretClass: simple-zookeeper-client-tls
+        "#;
+        let zookeeper: ZookeeperCluster = serde_yaml::from_str(input).expect("illegal test input");
+
+        println!("{}", serde_yaml::to_string(&zookeeper).unwrap());
+
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/crd/src/lib.rs]: Thanks done!

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -402,44 +404,32 @@ impl ZookeeperCluster {
...
     pub fn quorum_tls_secret_class(&self) -> String {
         let spec: &ZookeeperClusterSpec = &self.spec;
         spec.config
             .as_ref()
-            .and_then(|c| c.tls_config.as_ref())
-            .map(|tls| tls.quorum_tls_secret_class.as_ref())
-            .unwrap_or(DEFAULT_SECRET_CLASS)
+            .map(|c| c.quorum_tls_secret_class.as_ref())
+            .unwrap_or(TLS_DEFAULT_SECRET_CLASS)
             .to_string()
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/crd/src/lib.rs]: done

File: rust/crd/src/lib.rs [hunk 3]
```diff
@@ -55,49 +55,49 @@ pub struct ZookeeperClusterSpec {
...
-        }
-    }
+fn global_config_default() -> Option<GlobalZookeeperConfig> {
+    Some(GlobalZookeeperConfig {
+        tls: Some(TlsSecretClass {
+            secret_class: TLS_DEFAULT_SECRET_CLASS.to_string(),
+        }),
+        client_authentication: None,
+        quorum_tls_secret_class: TLS_DEFAULT_SECRET_CLASS.to_string(),
+    })
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/crd/src/lib.rs]: yeah thats better thanks!

**Review by siegfriedweber — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**